### PR TITLE
Add string format validation to `i18n-messages validate`

### PR DESCRIPTION
**Pull request by jimchamp**
Adds built-in string format validation to `i18n-messages validate`

### Technical

Restructures code by creating a directory of validator objects.  Each validator requires a Babel message and catalog in instantiation, and has a `validate` method that checks a single message and returns a list of error strings if message is invalid.

### Testing

1. Execute `i18n-messages validate` using a locale with known format string errors.
2. Ensure that error messages displayed identify all messages with errors.

### Stakeholders

@cdrini

### CODE REVIEW

File: openlibrary/i18n/validators/format_validator.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+from babel.messages.catalog import TranslationError
```
Comment (cdrini) [inline openlibrary/i18n/validators/format_validator.py:1]: This class hierarchy feels like too much here. The fact that both subclasses (1) don't override the constructor, and (2) have only one method, makes it look like these are just functions in disguise as classes. Let's dissolve the base class, and have two functions in a single `validators.py` file: `validate_python_format(message, catalog)` and `validate_not_fuzzy(message, catalog)`.

File: openlibrary/i18n/validators/format_validator.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        errors = []
+
+        if self.message.python_format:
+            try:
+                python_format(self.catalog, self.message)
+            except TranslationError as e:
+                if errors:
+                    errors.append(f'    {e}')
+                else:
+                    errors = [f'    {e}']
```
Comment (cdrini) [inline openlibrary/i18n/validators/format_validator.py:27]:
```suggestion
                errors.append(f'    {e}')
```

**Review by cdrini — CHANGES_REQUESTED** [review]
Nice! I can't believe this function already existed :D Haven't tested yet, but left some code quality feedback.

File: openlibrary/i18n/validators.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+
+def validate(message, catalog):
+    errors = _validate_fuzzy(message)
+    errors.extend(_validate_format(message, catalog))
+
+    return errors
+
+
+def _validate_format(message, catalog):
```
Comment (cdrini) [inline openlibrary/i18n/validators.py:12]: If you can find the type of these inputs, adding would be nice. Nice-to-have, not a blocker.

```suggestion
def _validate_format(message: Message, catalog: Catalog) -> List[str]:
```

**Review by cdrini — CHANGES_REQUESTED** [review]
- [ ] Run both `fmt` algorithms, since they seem to be catching different issues
- [ ] Pull in the format algorithm from test_po_file ; remove the `test_python_format` test in there.
- [ ] Make sure that format checker does not error for msgstr == "", and that it does not error for plurals (skip those)

File: openlibrary/i18n/validators.py [hunk 4]
```diff
@@ -42,3 +47,69 @@ def _validate_fuzzy(message):
             )
 
     return errors
+
+
+def _validate_cfmt(msgid: str, msgstr: str) -> List[str]:
+    errors = []
+
+    if len(msgstr) and isinstance(msgstr, str):
```
[the remark this answers is not in the document]
Comment (cdrini) [reply openlibrary/i18n/validators.py:55]: Ah, for plurals. Looks like this isn't working...

File: openlibrary/i18n/validators.py [hunk 5]
```diff
@@ -1,15 +1,20 @@
...
+
+from babel.messages.catalog import TranslationError, Message, Catalog
 from babel.messages.checkers import python_format
 
 
-def validate(message, catalog):
+def validate(message: Message, catalog: Catalog) -> List[str]:
     errors = _validate_fuzzy(message)
     errors.extend(_validate_format(message, catalog))
+    errors.extend(_validate_cfmt(message.id, message.string))
```
Comment (cdrini) [inline openlibrary/i18n/validators.py:12]: Thanks to the type hints, I can now quickly/easily investigate what other options these have!

```suggestion
    if message.string and message.pluralizeable:
        errors.extend(_validate_cfmt(message.id, message.string))
```

**Review by cdrini — CHANGES_REQUESTED** [review]

File: openlibrary/i18n/validators.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (jimchamp) [reply openlibrary/i18n/validators.py]: Replaced this with a test for `message.pluralizeable` in the `validate` function.

**Review by cdrini — APPROVED** [review]
Woohoo! This looks great :) Ran it over each language, and the output is _super_ handy!

**Pull request merged**